.github/workflows/macos.yml

```diff
@@ -0,0 +1,25 @@
+name: macOS
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  xcode:
+    runs-on: macos-10.15
+    strategy:
+      matrix:
+        xcode: [12.4, 12.3, 12.2, 12.1.1, 12.1, 12, 11.7, 11.6, 11.5, 11.4.1, 11.3.1, 11.2.1]
```

Comment:
To understand why this works, you need to know that xcode uses the "DEVELOPER_DIR" variable to select the version of xcode to use.